Makefile

```diff
@@ -83,17 +86,19 @@ assets/local/bosh-agent.exe: $(BOSH_AGENT_SOURCES)
 		cd -
 	mv $(BOSH_AGENT_REPO)/out/bosh-agent assets/local/bosh-agent.exe
 
-assets/local/bosh-blobstore-dav.exe: $(BOSH_AGENT_REPO)/integration/windows/fixtures/bosh-blobstore-dav.exe
-	@echo "### Creating/Updating assets/local/bosh-blobstore-dav.exe"
-	$(CP) $(BOSH_AGENT_REPO)/integration/windows/fixtures/bosh-blobstore-dav.exe assets/local
+assets/local/bosh-blobstore-dav.exe:
+	@echo "### Creating assets/local/bosh-blobstore-dav.exe"
+	$(eval BOSH_BLOBSTORE_DAV_FILE=$(shell curl -s $(BOSH_BLOBSTORE_DAV_URL) | xq -r '.ListBucketResult.Contents | last | .Key'))
```

Comment:
What version of `xq` was in use here? Latest version (1.2.4) doesn't seem to have a `-r` option. The closest equivalent is `-x` for XPath directives, which doesn't seem to support jq-style piping used here.

As an aside, when `xq` either isn't installed or returns an error, the make process continues, simply zipping up the result of the content returned from `curl` as the EXE file.

Reply:
Well that's interesting. I will take a look at this again on March 12th - trying it on a clean Linux system.

Reply:
@ystros There are two different `xq` tools. I apparently have the one installed with [yq](https://github.com/kislyuk/yq) which really just wraps `jq`.

> yq also supports XML. The yq package installs an executable, xq, which [transcodes XML to JSON](https://www.xml.com/pub/a/2006/05/31/converting-between-xml-and-json.html) using [xmltodict](https://github.com/martinblech/xmltodict) and pipes it to jq.

Not confusing at all.

I will attempt to use the version of `xq` that you get when you explicitly install `xq` via `brew install xq` or `apt-get install xq`.

Reply:
Both `xq`s don't seem to be a good solution.

1. s3 api does not order the results in a useful way so just selecting the last element doesn't select the latest version.
1. s3 doesn't allow you to specify a server side sort
1. xpath is terrible, especially at sorting

Let me think about this. I'm leaning towards configurable env vars with defaults.

Reply:
I moved the sorting outside xq, so it works now. I also tried it on Linux.